### PR TITLE
Add script to generate sources into generated branch

### PR DESCRIPTION
**Pull request by pkubatrh**
Created as a result of the discussion in https://github.com/sclorg/postgresql-container/pull/268

### CODE REVIEW

**Comment by pkubatrh** [comment]
@praiskup @omron93 ptal

**Comment by pkubatrh** [comment]
Made some more minor changes. Ready for review now.

File: update-generated.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# checkout source branch
+git checkout "$SOURCE_BRANCH" && git submodule update
+tag=$(git rev-parse HEAD)
+
+# copy the branch content into source/
+rsync -a ./* source
+VERSIONS=$(sed -n 's/^VERSIONS[[:space:]]*=//p' source/Makefile)
+
+git checkout "$GENERATED_BRANCH" && git submodule update
+for i in *; do
```
Comment (omron93) [inline update-generated.sh]: What about

```bash
shopt -s extglob
rm -rf !(common|sources)
```
?

File: update-generated.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    case "$i" in
+    # Do not remove the update script or common submodule
+    update|common|source)
+        continue
+        ;;
+    *)
+        rm -rf "$i"
```
Comment (omron93) [inline update-generated.sh]: I'm thinking... Why is needed to delete files in generated branch? - `rsync` could be used to update code by new code from `sources`.

 Without deleting files, `Add deleted files to the index as well` step isn't required.

Comment (pkubatrh) [reply update-generated.sh]: iirc we decided (while discussing source generation in general) that we do not want to keep archives of EOL images which is why everything gets removed first before the new generated sources get added.

File: update-generated.sh (hunk 1 above)
Comment (pkubatrh) [reply update-generated.sh]: There is still and update symlink in root that points to common scripts for sake of convenience.

Using extended globbing for this is a good idea though, thanks!

Comment (pkubatrh) [reply update-generated.sh]: >There is still an update symlink in root that points to common scripts for sake of convenience.

That is for the postgresql POC I will base from this PR at least.

**Comment by pkubatrh** [comment]
Thanks, merging

**Pull request merged**